### PR TITLE
Add FilePreviewOverride component

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

This PR adds a component to override the file preview for files that have the extension `demo`. The component renders the file's name and a button to close the preview. It requires the `onModalDismissed` prop introduced in https://github.com/mattermost/mattermost-webapp/pull/5430. Without the webapp PR, the close button does nothing, but you can still close the preview by clicking elsewhere or using the X button that appears on hover.

Another use of this component is to immediately close the file preview and trigger a separate action, which is why the `onModalDismissed` prop was introduced in the webapp PR. Should this go in the README if it is deemed useful? Here is an example:

```tsx
export function FilePreviewOverride(props: Props) {
    React.useEffect(() => {
        props.onModalDismissed();
        props.doSomethingElse();
    }, []);
    return null;
}
```

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-24763

#### Screenshots

When the attachment's name is `file.demo`

image

### CODE REVIEW

**Comment by hanzei** [comment]
Blocking until `v0.7.0` is released.

File: webapp/src/plugin.jsx [hunk 1]
```diff
@@ -192,6 +194,10 @@ export default class DemoPlugin {
         registry.registerAdminConsoleCustomSetting('SecretMessage', SecretMessageSetting, {showTitle: true});
         registry.registerAdminConsoleCustomSetting('CustomSetting', CustomSetting);
 
+        registry.registerFilePreviewComponent((fileInfo) => {
+            return isEnabled(store.getState()) && fileInfo.extension === 'demo';
```
Comment (marianunez) [inline webapp/src/plugin.jsx:198]: Is it necessary to check if the plugin is enabled? If the plugin is not enabled the webapp should not have custom preview component for that plugin

**Comment by marianunez** [comment]
> Another use of this component is to immediately close the file preview and trigger a separate action, which is why the `onModalDismissed` prop was introduced in the webapp PR. Should this go in the README if it is deemed useful? Here is an example:

Maybe this implementation with these extra details can be added to our documentation as a "How-to" [here](https://developers.mattermost.com/extend/plugins/best-practices/)

File: webapp/src/plugin.jsx (hunk 1 above)
Comment (mickmister) [reply webapp/src/plugin.jsx:198]: @marianunez I thought the `isEnabled` selector was pulling from the plugin's config in the system console to showcase a toggleable field, but I am mistaken. There is lots of code on the frontend and backend of this repo that is utilizing this value, when at no point should a code path reach a point where it is disabled. I think this is old code when the plugin wasn't auto-deactivated or something?

Will remove this check :+1:

**Comment by DHaussermann** [comment]
Actually, there is minor issue where the close button is not taking a style. 

ButtonStyle

@mickmister is this a change we need here or in the corresponding webapp PR https://github.com/mattermost/mattermost-webapp/pull/5430

**Comment by mickmister** [comment]
@DHaussermann The issue is in this PR. I did not spend enough time to make the component look decent. I'll give it another go and take the user's theme into account.

**Comment by mickmister** [comment]
@DHaussermann Here's the style now 🙂 

image

**Comment by DHaussermann** [comment]
Thanks @mickmister :)
- Styling on dark theme now colors the button in a way that is consistent with other buttons on user's dark theme. 
- Spacing around the button has also been fixed

Cosmetic issues have now been resolved. 
LGTM!

**Comment by DHaussermann** [comment]
@hanzei I have left the `Do Not Merge` label on. 
Please advice when it's appropriate to merge this into Master branch ad `v0.7.0` has now shipped.

**Pull request merged**